### ISSUE TITLE
Using a Texture

### ISSUE THREAD

**Issue by theMK2k** [XT95/VisualLiveSystem#2]
Hi XT95,

I'm currently "remixing" a scene from Shadertoy (https://www.shadertoy.com/view/4s3SRN) to be used in Visual Live Sytem.

I got some things parametrized with the CC inputs, bassTime, time and bass, also added the godrays layer of your Abstraction Dominante scene for some nice strobe effects.

Unfortunately I can't get it to work with a texture. The original source uses the iChannel0 variable for it. So I added the following line to config.xml:

`<channel type="image" id="iChannel0" value="noise256.png"/>`

It doesn't seem to have any effect, though. Any help would be much appreciated. I added the current source to this issue.

If you want to test it in VLS, set sliders CC[0] and/or CC[1] greater than 0, else you see a black screen.

[ST Fractal Flythrough.zip](https://github.com/XT95/VisualLiveSystem/files/1756038/ST.Fractal.Flythrough.zip)

**Comment by XT95** [comment]
Hi !
Glad to see someone use VLS besides me :)

So I checked your shader and the problem isn't the loading of the texture.
If you put  fragColor = texture2D(iChannel0, v); at the end of void
mainImage(...) you should see the noise.
The problem are in the end of the function tex3D() :
  //return (texture(tex, p.yz)*n.x + texture(tex, p.zx)*n.y + texture(tex,
p.xy)*n.z).xyz;
  return vec3 (1.,1.,1.);


You are just returning white color here :D Maybe you did that when you port
the code from Shadertoy to visual live system because texture() is unknow
in GLSL 2.1, it's texture2D() !
So if you put : return (texture2D(tex, p.yz)*n.x + texture2D(tex, p.zx)*n.y
+ texture2D(tex, p.xy)*n.z).xyz; instead of return vec3(1.,1.,1.), it works
well !

Do you use VLS for a VJ performance ? Do you have a facebook page or a
twitter account where I can follow you :) ?

Have good week,
Anatole



2018-02-25 14:09 GMT+01:00 MK2k <notifications@github.com>:

**Comment by theMK2k** [comment]
Hey Anatole,

wow thanks. Yeah, it seems I commented the "return texture ... " statement out because in the first step of porting I ran into errors :D

Now, I'm currently quite a noob when it comes to GLSL and for the next weeks and months I will most likely try to port some great Shadertoy stuff to VLS and ultimately provide the results to the http://aduprat.com/vls/ site if the shader's license permits ;) I'll build up a porting documentation along the way, too.

After building up a nice collection of shaders I'm planning on doing a VJ performance probably alongside mixing music. I had an FM radio show for 5 years in Germany at a community run radio. Some of the archived mixes can be seen here: https://rewind-show.net/?page_id=2

Now because of my new born child I'm more often at home, so it's a perfect situation to get started in VLS, which I followed since the initial release in 2012.

Apart from music, I have some (pretty standard) code projects my own. My twitter is https://twitter.com/theMK2k

cheers

-- Jörg / MK2k
